pkg/controller/periodic/job.go

```diff
@@ -160,3 +160,57 @@ func (j *JobController) WaitForJobCompletion(ctx context.Context, job *batchv1.J
 		}
 	}
 }
+
+func (j *JobController) createVolumeSource(ctx context.Context, storage *insightsv1alpha1.Storage) corev1.VolumeSource {
+	if storage == nil {
+		klog.Info("Creating volume source with EmptyDir, no storageSpec provided")
+		return corev1.VolumeSource{
+			EmptyDir: &corev1.EmptyDirVolumeSource{},
+		}
+	}
+
+	if storage.Type == insightsv1alpha1.StorageTypePersistentVolume {
+		// Validate if the PVC exists
+		persistentVolumeClaimName := storage.PersistentVolume.Claim.Name
+		pvc, err := j.kubeClient.CoreV1().PersistentVolumeClaims(insightsNamespace).Get(ctx, persistentVolumeClaimName, metav1.GetOptions{})
+		if err != nil {
+			klog.Error(err, " Failed to get PersistentVolumeClaim with name ", persistentVolumeClaimName)
+		} else if pvc != nil {
+			klog.Infof("Creating volume source with PersistentVolumeClaimName: %s", persistentVolumeClaimName)
+			return corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: persistentVolumeClaimName,
+				},
+			}
+		}
+	}
+
+	klog.Info("Creating volume source with EmptyDir")
+	return corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
+	}
+}
```

Comment:
I think this function would deserve some comment/documentation - especially the fact that in case of an error the `EmptyDirVolumeSource` is used. I was thinking if failing the job would be better (in case of an error with PVC) but I don't have strong opinion.

Reply:
Added a comment for the function, hopefully it is sufficient. 

I think logging the error message when failing to get the PVC and then using an ephemeral volume instead should be enough, but I am open to changing it if needed